ldt/src/quotient.rs

```diff
@@ -181,77 +286,212 @@ impl<F, EF, Inner: MatrixRowSlices<F>> Matrix<EF> for QuotientMatrix<F, EF, Inne
     }
 }
 
-impl<F: Field, EF: ExtensionField<F>, Inner: MatrixRowSlices<F>> MatrixRows<EF>
-    for QuotientMatrix<F, EF, Inner>
-{
-    type Row<'a> = QuotientMatrixRow<'a, F, EF> where Inner: 'a;
+impl<F: Field, Inner: MatrixRowSlices<F>> MatrixRows<F> for QuotientMatrix<F, Inner> {
+    type Row<'a> = Vec<F> where Inner: 'a;
 
     #[inline]
     fn row(&self, r: usize) -> Self::Row<'_> {
-        let num_openings = self.openings.len();
-        QuotientMatrixRow {
-            openings: &self.openings,
-            inv_denominator: &self.inv_denominators[r * num_openings..(r + 1) * num_openings],
-            inner_row: self.inner.row_slice(r),
-            opening_index: 0,
-            inner_col_index: 0,
-        }
+        compute_quotient_matrix_row(
+            self.subgroup[r],
+            &self.openings,
+            self.inv_denominators.row_slice(r),
+            self.inner.row_slice(r),
+        )
     }
 }
 
-pub struct QuotientMatrixRow<'a, F, EF> {
-    openings: &'a [Opening<EF>],
-    /// `1 / (x - opened_point)`
-    inv_denominator: &'a [EF],
-    inner_row: &'a [F],
-    opening_index: usize,
-    inner_col_index: usize,
-}
+fn compute_quotient_matrix_row<F: Field>(
+    x: F,
+    openings: &[Opening<F>],
+    m_invs: &[F],
+    inner_row: &[F],
+) -> Vec<F> {
+    let mut qp_ys = Vec::with_capacity(inner_row.len() * openings.len());
 
-impl<'a, F, EF> Iterator for QuotientMatrixRow<'a, F, EF>
-where
-    F: Field,
-    EF: ExtensionField<F>,
-{
-    type Item = EF;
+    // this is always EF::D.
+    let r_poly_len = openings[0].remainder_polys[0].len();
+    // [P(1,1,1,1),P(x,x,x,x),P(x^2,x^2,x^2,x^2),..]
+    let packed_x_pows = x
+        .powers()
+        .take(r_poly_len)
+        .map(|x_pow| F::Packing::from(x_pow))
+        .collect_vec();
 
-    #[inline]
-    fn next(&mut self) -> Option<Self::Item> {
-        if self.inner_col_index == self.inner_row.len() {
-            self.opening_index += 1;
-            self.inner_col_index = 0;
+    for (opening, &m_inv) in openings.iter().zip(m_invs) {
+        let packed_m_inv = F::Packing::from(m_inv);
+
+        let process_scalar = |y: F, r: &[F], qp_ys: &mut Vec<F>| {
+            qp_ys.push((y - eval_poly(r, x)) * m_inv);
+        };
+
+        let (pfx_ys, packed_ys, sfx_ys) = unsafe { inner_row.align_to::<F::Packing>() };
+
+        // this is usually true, if the inner MMCS is holding them in a Vec,
+        // and the system allocator default alignment is greater than F::Packing's alignment.
+        // For example, with an inner DirectMmcs, default allocator, and 16 byte packing.
+        // It isn't true, for example, with jemalloc, which can hand out 8 byte alignments.
+        // If that's the case, this code will slow down greatly and we will need to reconsider.
+
+        if !pfx_ys.is_empty() {
```

Comment:
this is so sad

We probably do, in fact, want jemalloc, so need to figure out what to do here, otherwise it will never do simd